HousingSearchApi/V2/Gateways/SearchGateway.cs

```diff
@@ -17,22 +18,49 @@ public SearchGateway(IElasticClient elasticClient)
     public async Task<SearchResponseDto> Search(string indexName, SearchParametersDto searchParams)
     {
         var searchResponse = await _elasticClient.SearchAsync<object>(s => s
-                .Index(indexName)
-                .Query(q => q
-                    .SimpleQueryString(qs => qs
-                        .Fields("*")
-                        .Query(searchParams.SearchText)
+            .Index(indexName)
+            .Query(q => q
+                .Bool(b => b
+                    .Should(
+                        MultiMatchSingleField(searchParams.SearchText, boost: 6),
+                        MultiMatchAcrossFields(searchParams.SearchText, boost: 2)
                     )
                 )
-                .Size(searchParams.PageSize)
-                .From((searchParams.PageNumber - 1) * searchParams.PageSize)
-                .TrackTotalHits(true)
+            )
+            .MinScore(15)
+            .Size(searchParams.PageSize)
+            .From((searchParams.PageNumber - 1) * searchParams.PageSize)
```

Comment:
The page number is validated to be 1+, right?
Same with pagesize to be at least 1+

Reply:
Yep